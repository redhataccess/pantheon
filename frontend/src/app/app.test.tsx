import React from 'react';
import Search from '@app/search';
import "isomorphic-fetch"

import { mount, shallow } from 'enzyme';
import { Label } from '@patternfly/react-core';

describe('App tests', () => {
  test('should render default App component', () => {
    const view = shallow(<Search />);
    expect(view).toMatchSnapshot();
  });

  it('should render a search Label', () => {
    const wrapper = mount(<Search />);
    const button = wrapper.find(Label);
    expect(button.exists()).toBe(true)
  });

});
