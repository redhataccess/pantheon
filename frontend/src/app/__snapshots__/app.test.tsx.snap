// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<Fragment>
  <div>
    <div>
      <div
        className="row-view"
      >
        <Label
          className=""
          isCompact={false}
        >
          Search Query:
        </Label>
        <TextInput
          aria-label={null}
          className=""
          id="search"
          isDisabled={false}
          isReadOnly={false}
          isRequired={false}
          isValid={true}
          onChange={[Function]}
          onKeyDown={[Function]}
          type="text"
          value="*"
        />
        <Label
          className=""
          isCompact={false}
        >
          Start At:
        </Label>
        <TextInput
          aria-label={null}
          className=""
          id="pageNum"
          isDisabled={false}
          isReadOnly={false}
          isRequired={false}
          isValid={true}
          onChange={[Function]}
          onKeyDown={[Function]}
          pattern="[0-9]*"
          type="text"
          value={1}
        />
        <Label
          className=""
          isCompact={false}
        >
          Result Count:
        </Label>
        <TextInput
          aria-label={null}
          className=""
          id="pageCount"
          isDisabled={false}
          isReadOnly={false}
          isRequired={false}
          isValid={true}
          onChange={[Function]}
          onKeyDown={[Function]}
          pattern="[0-9]*"
          type="text"
          value={50}
        />
        <Button
          aria-label={null}
          className=""
          component="button"
          isActive={false}
          isBlock={false}
          isDisabled={false}
          isFocus={false}
          isHover={false}
          isInline={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Search
        </Button>
      </div>
      <div
        className="notification-container"
      >
        <Alert
          action={null}
          className=""
          title="Search is case sensitive. Type '*' and press 'Enter' for all the modules."
          variant="info"
          variantLabel={null}
        />
      </div>
      <DataList
        aria-label="Simple data list example"
        className=""
      >
        <DataListItem
          aria-labelledby="simple-item1"
          className=""
          isExpanded={false}
        >
          <DataListItemRow
            className=""
            id="data-rows-header"
          >
            <DataListCheck
              aria-label="controlled checkbox example"
              aria-labelledby="width-ex1-check1"
              className=""
              id="check"
              isChecked={false}
              isDisabled={false}
              isValid={true}
              onChange={[Function]}
              onClick={[Function]}
            />
            <DataListItemCells
              className=""
              dataListCells={
                Array [
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    onClick={[Function]}
                    width={2}
                  >
                    <span
                      className="sp-prop"
                      id="span-name"
                    >
                      Name
                    </span>
                  </DataListCell>,
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    onClick={[Function]}
                    width={2}
                  >
                    <span
                      className="sp-prop"
                      id="span-description"
                    >
                      Description
                    </span>
                  </DataListCell>,
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    width={1}
                  >
                    <span
                      className="sp-prop-nosort"
                      id="span-source-type"
                    >
                      Source Type
                    </span>
                  </DataListCell>,
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    width={1}
                  >
                    <span
                      className="sp-prop-nosort"
                      id="span-source-name"
                    >
                      Source Name
                    </span>
                  </DataListCell>,
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    onClick={[Function]}
                    width={1}
                  >
                    <span
                      className="sp-prop"
                      id="span-upload-time"
                    >
                        Upload Time
                    </span>
                  </DataListCell>,
                ]
              }
            />
          </DataListItemRow>
          <DataListItemRow
            className=""
            id="data-rows"
          >
            <DataListCheck
              aria-label="controlled checkbox example"
              aria-labelledby="width-ex3-check1"
              className=""
              id=""
              isChecked={false}
              isDisabled={false}
              isValid={true}
              name=""
              onChange={[Function]}
              onClick={[Function]}
            />
            <DataListItemCells
              className=""
              dataListCells={
                Array [
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    width={2}
                  >
                    <span>
                      
                    </span>
                  </DataListCell>,
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    width={2}
                  >
                    <span>
                      No items found to be displayed
                    </span>
                  </DataListCell>,
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    width={1}
                  >
                    <span />
                  </DataListCell>,
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    width={1}
                  >
                    <span>
                      
                    </span>
                  </DataListCell>,
                  <DataListCell
                    alignRight={false}
                    className=""
                    isFilled={true}
                    isIcon={false}
                    width={1}
                  >
                    <span>
                      
                    </span>
                  </DataListCell>,
                ]
              }
              key=""
              onClick={[Function]}
            />
          </DataListItemRow>
          <DataListItemRow
            className=""
            id="data-rows"
          />
        </DataListItem>
      </DataList>
      <div
        className="alert"
      />
    </div>
  </div>
</Fragment>
`;
